packages/nuxt/src/runtime/plugins/formkit-validation.server.ts

```diff
@@ -0,0 +1,43 @@
+import { defineNuxtPlugin, useRequestEvent } from '#imports'
+import type { FormKitNode } from '@formkit/core'
+import { stopWatch, watchRegistry } from '@formkit/core'
+import { getValidationMessages } from '@formkit/validation'
+
+export default defineNuxtPlugin((nuxtApp) => {
+  const context = useRequestEvent()?.context?.formkit
+  if (!context || !context._id) return
+
+  let node: FormKitNode
+
+  // This will be called when the node is initialised
+  const unsub = watchRegistry(context._id, (event) => {
+    if (event.payload) {
+      node = event.payload
+      node.input(context._data)
```

Comment:
@danielroe this is the async culprit. By default all `node.input(data)` calls in FormKit are async. There is a second param (`async = true` by default) to make this not synchronous. I’ll clean this up on my end after merging.